### PR TITLE
Fixes #31093 - Remove Config Groups UI

### PR DESCRIPTION
**Pull request by shiramax**
Extracted in theforeman/foreman_puppet_enc#14.

### CODE REVIEW

File: db/seeds.d/020-permissions_list.rb [hunk 1]
```diff
@@ -34,10 +34,6 @@ def permissions
         ['ConfigReport', 'view_config_reports'],
         ['ConfigReport', 'destroy_config_reports'],
         ['ConfigReport', 'upload_config_reports'],
-        ['ConfigGroup', 'view_config_groups'],
-        ['ConfigGroup', 'create_config_groups'],
-        ['ConfigGroup', 'edit_config_groups'],
-        ['ConfigGroup', 'destroy_config_groups'],
```
Comment (ezr-ondrej) [inline db/seeds.d/020-permissions_list.rb]: Can we separate PR to drop the permissions?
We will need to drop them from https://github.com/theforeman/foreman/blob/324bafdb625b06fe939530ef77534170be855447/app/registries/foreman/access_permissions.rb#L267 as well and we can't do it before both UI and API are in.

**Review by ezr-ondrej — COMMENTED** [review]
`puppetclasses/_class_selection` needs the template `config_groups/_config_groups_selection` and tests are failing because of that. Can we just move that template to the `puppetclasses/_config_groups_selection` (so we don't forget to remove, but tests would pass)

File: app/registries/menu/loader.rb [hunk 2]
```diff
@@ -74,7 +74,7 @@ def self.load
           menu.divider                  :caption => N_('Puppet')
           menu.item :environments,      :caption => N_('Environments')
           menu.item :puppetclasses,     :caption => N_('Classes')
-          menu.item :config_groups,     :caption => N_('Config Groups')
+          menu.item :puppetclass_lookup_keys, :caption => N_('Smart Class Parameters')
```
Comment (ezr-ondrej) [inline app/registries/menu/loader.rb]: This has been already dropped in previous commit in this branch, we should keep it removed :+1: 
```suggestion
          menu.item :puppetclass_lookup_keys, :caption => N_('Smart Class Parameters')
```

**Review by ezr-ondrej — CHANGES_REQUESTED** [review]
One issue comming from a bad rebase probably.
And the failing test seems to need `config_groups/config_group` sorry I didn't realize that.
Not sure what to do about that. We could just wait with merge of this till the Puppetclasses are extracted.
Other tests seems to be passing, so it should pass after the Puppetclasss UI extraction. I guess it's the best approach here.

In that case, you don't need to move the `config_groups_selection` and we just wait with this a while :)

**Comment by ezr-ondrej** [comment]
Although https://github.com/ezr-ondrej/foreman/pull/8 needs to go first, right?

File: app/registries/menu/loader.rb (hunk 2 above)
Comment (shiramax) [reply app/registries/menu/loader.rb]: done

**Comment by shiramax** [comment]
@ezr-ondrej yes, I think the right order is 
- Controllers 
- Models 
- UI
-API
-Permissions

**Review by ezr-ondrej — APPROVED** [review]
:heart:

**Pull request merged**